io/subsystem/src/main/java/org/wildfly/extension/io/BufferPoolAdd.java

```diff
@@ -0,0 +1,41 @@
+package org.wildfly.extension.io;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP_ADDR;
+
+import org.jboss.as.controller.AbstractAddStepHandler;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.dmr.ModelNode;
+import org.jboss.msc.service.ServiceController;
+
+class BufferPoolAdd extends AbstractAddStepHandler {
+
+    static final BufferPoolAdd INSTANCE = new BufferPoolAdd();
+
+    BufferPoolAdd() {
+        super(BufferPoolResourceDefinition.ATTRIBUTES);
+    }
+
+    @Override
+    protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model)
+            throws OperationFailedException {
+        final PathAddress address = PathAddress.pathAddress(operation.get(OP_ADDR));
+        final String name = address.getLastElement().getValue();
```

Comment:
You can directly use `org.jboss.as.controller.OperationContext#getCurrentAddressValue`